pkg/controller/nodelifecycle/scheduler/timed_workers.go

```diff
@@ -52,11 +52,17 @@ type TimedWorker struct {
 // createWorker creates a TimedWorker that will execute `f` not earlier than `fireAt`.
 func createWorker(ctx context.Context, args *WorkArgs, createdAt time.Time, fireAt time.Time, f func(ctx context.Context, args *WorkArgs) error, clock clock.WithDelayedExecution) *TimedWorker {
 	delay := fireAt.Sub(createdAt)
+	fWithErrorLogging := func() {
+		err := f(ctx, args)
+		if err != nil {
+			klog.Errorf("Timed worker failed with error: %q", err)
```

Comment:
Sorry for last minute suggestion, but thinking of debugability, it's probably worth saying something like:

```suggestion
			klog.Errorf("NodeLifecycle: timed worker failed with error: %q", err)
```